src/dbus_api/pool.rs

```diff
@@ -171,25 +172,31 @@ fn add_devs(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 
     let blockdevs = devs.map(|x| Path::new(x)).collect::<Vec<&Path>>();
 
-    let msg = match pool.add_blockdevs(&blockdevs, force) {
-        Ok(devnodes) => {
-            let paths = devnodes
+    let result = pool.add_blockdevs(&blockdevs, force);
+    let msg = match result {
+        Ok(ref infos) => {
+            let mut return_value = Vec::new();
+            for uuid in infos {
+                let bd_object_path: dbus::Path =
+                    create_dbus_blockdev(dbus_context, object_path.clone(), *uuid);
```

Comment:
Might as well be consistent w/ above, wrap ```create_dbus_blockdev...``` w/ ```MessageItem::ObjectPath``` call as done in api.rs and drop line that starts at 185.

Reply:
yep, ok.